### PR TITLE
LPD-42544 Create a sample for a FDS View component so that it can be tested

### PR DESCRIPTION
**Pull request by chestofwonders**
WIP POC for https://liferay.atlassian.net/browse/LPD-42544 (parent task)

Built over https://github.com/liferay-frontend/liferay-portal/pull/4571 to add a Carousel component

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx [hunk 1]
```diff
@@ -3,14 +3,99 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-import React from 'react';
+import React, {useState} from 'react';
+import {Card} from '../../../../../../../frontend-data-set-web/src/main/resources/META-INF/resources/views/cards/Cards';
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx:7]: This is not the right way to import. It should be `import {Card} from '@liferay/frontend-data-set-web';`, along with export from frontend-data-set-web module.

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx [hunk 2]
```diff
@@ -3,14 +3,99 @@
...
+import './carousel.scss';
 
 const CarouselView = (props: any) => {
-
-	// eslint-disable-next-line no-console
-	console.log('props = ', props);
-
-	return <>Add real implementation here.</>;
+	const initialItems = Array.from({ length: 16 }, (_, index) => ({
+		url: `https://via.placeholder.com/800x500?text=Imagen+${index + 1}`,
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx:12]: Even in samples, we should not link to real external sites. If you need images, some ship with DXP, as samples in document library. See how we use them in [ckeditor sample module](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web/src/main/resources/META-INF/resources/partials/balloon.jsp#L23).

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx [hunk 3]
```diff
@@ -3,14 +3,99 @@
...
+import {Card} from '../../../../../../../frontend-data-set-web/src/main/resources/META-INF/resources/views/cards/Cards';
+import './carousel.scss';
 
 const CarouselView = (props: any) => {
-
-	// eslint-disable-next-line no-console
-	console.log('props = ', props);
-
-	return <>Add real implementation here.</>;
+	const initialItems = Array.from({ length: 16 }, (_, index) => ({
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx:11]: What is this black magic? We are converting `{ length: 16 }` object to an array? This looks like something generated by ChatGPT :sweat_smile:

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx [hunk 4]
```diff
@@ -3,14 +3,99 @@
...
+		  image: 'image',
+		  link: 'link',
+		  sticker: 'sticker',
+		  symbol: 'symbol',
+		  title: 'title',
+		};
+  
+	  const [selectedIndex, setSelectedIndex] = useState(0);
+	
+	  const thumbnailsToShow = 5;
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx:28]: `visibleThumbnailsCount`?

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx [hunk 5]
```diff
@@ -3,14 +3,99 @@
...
+		setSelectedIndex(newIndex);
+	  };
+	
+	  const handleNextClick = () => {
+		const newIndex =
+		  selectedIndex === initialItems.length - 1 ? 0 : selectedIndex + 1;
+		setSelectedIndex(newIndex);
+	  };
+	
+	  const { url, name, size } = initialItems[selectedIndex];
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx:53]: Are you using VSCode? If you do, please install `Prettier` extension, and have it run on each save. It'll automatically take care of 95% of SF issues.

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/carousel.scss [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    width: 100%;
+    background-color: black;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    position: relative;
+  }
+  
+  .nav-buttons {
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/carousel.scss:34]: All the classes in this CSS file are in root and have generic names, so they are very likely to spill over to the rest of the page. Please put the implementation, along with all the styles, into a very specific container, named something like `fds-carousel-view-sample`.

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/carousel.scss
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/carousel.scss:1]: To follow typical placement and naming convention for css files, it should be `/resources/css/CarouselView.scss`.

**Review by markocikos — CHANGES_REQUESTED** [review]
@chestofwonders Please see comments inline

File: modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx (hunk 1 above)
Comment (chestofwonders) [reply modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/js/CarouselView.tsx:7]: That will be my next question, thank you!

**Comment by markocikos** [comment]
Followup in https://github.com/liferay-frontend/liferay-portal/pull/4576

**Pull request closed without merging**